packages/opentelemetry-core/src/utils/environment.ts

```diff
@@ -16,33 +16,81 @@
 
 import { LogLevel } from '../common/types';
 
-export type ENVIRONMENT_MAP = { [key: string]: string | number };
+const DEFAULT_LIST_SEPARATOR = ',';
 
 /**
  * Environment interface to define all names
  */
-export interface ENVIRONMENT {
-  OTEL_LOG_LEVEL?: LogLevel;
-  OTEL_NO_PATCH_MODULES?: string;
+
+type ENVIRONMENT_NUMBERS = {
   OTEL_SAMPLING_PROBABILITY?: number;
   OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT?: number;
   OTEL_SPAN_EVENT_COUNT_LIMIT?: number;
   OTEL_SPAN_LINK_COUNT_LIMIT?: number;
-}
+};
 
-const ENVIRONMENT_NUMBERS: Partial<keyof ENVIRONMENT>[] = [
+const ENVIRONMENT_NUMBERS_KEYS: (keyof ENVIRONMENT_NUMBERS)[] = [
   'OTEL_SAMPLING_PROBABILITY',
   'OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT',
   'OTEL_SPAN_EVENT_COUNT_LIMIT',
   'OTEL_SPAN_LINK_COUNT_LIMIT',
 ];
 
+function isEnvVarANumber(key: unknown): key is keyof ENVIRONMENT_NUMBERS {
+  return (
+    ENVIRONMENT_NUMBERS_KEYS.indexOf(key as keyof ENVIRONMENT_NUMBERS) > -1
+  );
+}
+
+type ENVIRONMENT_LISTS = {
+  OTEL_NO_PATCH_MODULES?: string[];
+};
+
+const ENVIRONMENT_LISTS_KEYS: (keyof ENVIRONMENT_LISTS)[] = [
```

Comment:
and the same here ?
```
const ENVIRONMENT_LISTS_KEYS = Object.keys(ENVIRONMENT_LISTS);
```

Reply:
Without the keyof typecast, the ENVIRONMENT_LISTS_KEYS var gets the type string[]

Reply:
I'm just talking about having the Object.keys instead of manually listing the same thing again, but it should still be possible to do `as key of .....etc.`

Reply:
`Object.keys(ENVIRONMENT_NUMBERS)` yields error:
> 'ENVIRONMENT_NUMBERS' only refers to a type, but is being used as a value here.
If you really want to not duplicate keys, then we can start the typing hierarchy with a defaults object, and extract all type definitions from it.